apps/contributions/tests/views/test_switchboard.py

```diff
@@ -615,6 +615,70 @@ def test_requests_when_token_expired(self, api_client, switchboard_api_expired_t
         assert response.status_code == status.HTTP_401_UNAUTHORIZED
         assert response.json() == {"detail": "Invalid token."}
 
+    def test_filter_by_provider_subscription_id(self, api_client, switchboard_api_token):
+        subscription_contribution = ContributionFactory(monthly_subscription=True)
+        ContributionFactory(monthly_subscription=True)
+        ContributionFactory(monthly_subscription=True)
+
+        response = api_client.get(
+            reverse("switchboard-contribution-list"),
+            {"provider_subscription_id": subscription_contribution.provider_subscription_id},
+            headers={"Authorization": f"Token {switchboard_api_token}"},
+        )
+        assert response.status_code == status.HTTP_200_OK
+        results = response.json()["results"]
+        assert len(results) == 1
```

Comment:
minor, but this can be a single line

```suggestion
        assert len((results:=response.json()["results"])) == 1
```

Reply:
Nice, this is elegant https://github.com/newsrevenuehub/rev-engine/pull/1745/commits/632fc501782984c6e04d11702e36142e7d4f139f